app/src/main/java/com/infomaniak/drive/ui/MainViewModel.kt

```diff
@@ -318,7 +323,18 @@ class MainViewModel(
                     runCatching { localFolder.children.remove(file) }
                 }
 
-                FileController.addChild(newParent.id, file.apply { parentId = newParent.id }, realm)
+                if (isSharedWithMe) {
+                    // If the selected folder is a shared folder, it is removed from the user's realm table (userId_driveId)
+                    // and added to the share realm (userId_share)
+                    FileController.removeFile(fileId = file.id, customRealm = realm)
+                    FileController.getRealmInstance(UserDrive(sharedWithMe = true)).use { realm ->
```

Comment:
Name this realm "sharedRealm" to be sure to avoid confusion with the realm just above (that could be renamed to "currentDriveRealm" or something like this